### PR TITLE
Twitter

### PR DESCRIPTION
**Pull request by MichaelPetrinolis**
Sign in with your twitter account

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Twitter/Views/TwitterLoginSettings.Edit.cshtml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+@using OrchardCore.Twitter.ViewModels;
+@model TwitterLoginSettingsViewModel;
+
+<p class="alert alert-warning">@T["The current tenant will be reloaded when the settings are saved."]</p>
+
+<h3>@T["Twitter Login Settings"]</h3>
+
+<fieldset class="form-group" asp-validation-class-for="ConsumerKey">
+    <label>@T["ConsumerKey"] <span asp-validation-for="ConsumerKey"></span></label>
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Twitter/Views/TwitterLoginSettings.Edit.cshtml]:
```suggestion
    <label asp-for="ConsumerKey">@T["ConsumerKey"] <span asp-validation-for="ConsumerKey"></span></label>
```

File: src/OrchardCore.Modules/OrchardCore.Twitter/Views/TwitterLoginSettings.Edit.cshtml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    <input asp-for="ConsumerKey" class="form-control" />
+    <span class="hint">@T["The Consumer Key defined in the Keys and tokens tab of twitter app dashboard."]</span>
+</fieldset>
+
+<fieldset class="form-group" asp-validation-class-for="ConsumerSecret">
+    <label>@T["Secret Key"] <span asp-validation-for="ConsumerSecret"></span></label>
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Twitter/Views/TwitterLoginSettings.Edit.cshtml]:
```suggestion
    <label asp-for="ConsumerSecret">@T["Secret Key"] <span asp-validation-for="ConsumerSecret"></span></label>
```

File: src/OrchardCore.Modules/OrchardCore.Twitter/Views/TwitterLoginSettings.Edit.cshtml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    <input asp-for="ConsumerSecret" class="form-control" type="password" value="@Model.ConsumerSecret" />
+    <span class="hint">@T["The Consumer Secret defined in the Keys and tokens tab of twitter app dashboard."]</span>
+</fieldset>
+
+<fieldset class="form-group" asp-validation-class-for="CallbackPath">
+    <label>@T["CallbackPath"] <span asp-validation-for="CallbackPath"></span></label>
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Twitter/Views/TwitterLoginSettings.Edit.cshtml]:
```suggestion
    <label  asp-for="CallbackPath" >@T["CallbackPath"] <span asp-validation-for="CallbackPath"></span></label>
```

File: src/OrchardCore.Modules/OrchardCore.Twitter/Startup.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    public class Startup : StartupBase
+    {
+        public override void ConfigureServices(IServiceCollection services)
+        {
+            services.AddScoped<IPermissionProvider, Permissions>();
+        }
+    }
+
+    [Feature(TwitterConstants.Features.TwitterLogin)]
+    public class MicrosoftAccountStartup : StartupBase
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Twitter/Startup.cs]:
```suggestion
    public class TwitterLoginStartup : StartupBase
```

**Comment by MichaelPetrinolis** [comment]
I am also changing the class names from TwitterLogin to TwitterSignin to be more consistent with twitter naming

**Pull request merged**